hacking/build_library/build_ansible/command_plugins/plugin_formatter.py

```diff
@@ -153,16 +153,17 @@ def rst_xline(width, char="="):
 def documented_type(text):
     ''' Convert any python type to a type for documentation '''
 
-    if isinstance(text, Undefined):
-        return '-'
-    if text == 'str':
-        return 'string'
-    if text == 'bool':
-        return 'boolean'
-    if text == 'int':
-        return 'integer'
-    if text == 'dict':
-        return 'dictionary'
+    # list/path/etc are already the same as their description
+    # we only put here those that need 'long name translation'.
+    if text == 'str' or isinstance(text, Undefined):
+        text = 'string'
```

Comment:
We don't want to say an option is a **string** if that is not true. Most modules have no type-information for the docs, so that means suddenly all parameters become strings.

That is the reason why I proposed it as part of: https://github.com/ansible/ansible/pull/45805

But only part of that PR was accepted, which meant that if you wanted type-information in the docs, you have to provide it separately.

In any case, this change is wrong, and *even* if we could use the argspec in the future, there will always be a need to document the type differently then the argspec (i.e. in case of raw for backward compatibility).

Reply:
Also see: https://github.com/ansible/ansible/pull/56312